src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,359 @@
+import {
+  DEFAULT_EXPRESSION,
+  LEADING_HYPHEN,
+  NAME_EXPRESSION,
+  RETURN_EXPRESSION,
+  TYPE_EXPRESSION,
+} from '../constants.mjs';
+import { buildHierarchy } from './buildHierarchy.mjs';
+import parseSignature from './parseSignature.mjs';
+import { getRemarkRehype } from '../../../utils/remark.mjs';
+import { transformNodesToString } from '../../../utils/unist.mjs';
+
+const sectionTypePlurals = {
+  module: 'modules',
+  misc: 'miscs',
+  class: 'classes',
+  method: 'methods',
+  property: 'properties',
+  global: 'globals',
+  example: 'examples',
+  // Constructors should go under a class sections' signatures property
+  ctor: 'signatures',
+  classMethod: 'classMethods',
+  event: 'events',
+  var: 'vars',
+};
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @returns {import('../types.d.ts').Meta | undefined}
+ */
+function createMeta(entry) {
+  const arrify = val => (Array.isArray(val) ? val : [val]);
```

Comment:
IMO we can find a better name for this function 🙇 JSDoc parameters and return type are needed

Reply:
`enforceArray`? `ensureArray`? `toArray`?

Reply:
`enforceArray` +1

Reply:
+1 for `enforceArray`